Makefile

```diff
@@ -1,22 +1,75 @@
 
 # Test tshark's dissectors on small test files
 
-TEST_CASES = $(wildcard tests/*/*.pdml)
+# List all available test directories
+TEST_CASE_DIRS=$(wildcard tests/*/*)
+
+# Convert directories to test case names
+TEST_CASES=$(foreach test,$(TEST_CASE_DIRS), $(test)/$(notdir $(test)))
+
+# List of versions for which we check and store different outputs
+# When output is verified, current wireshark version's output is compared to same version's stored output or to the latest previous version
+# - list should be ordered from the oldest to the newest version
+SUPPORTED_VERSIONS?=2.0 2.2 2.3
+VERSION?=
+SELECTED_VERSIONS=$(if $(VERSION),$(VERSION),$(SUPPORTED_VERSIONS))
 
 TSHARK_EXECUTABLE?=tshark
+TSHARK_VERSION=$(shell $(TSHARK_EXECUTABLE) --version | head -1 | cut -d' ' -f 3 | cut -d'.' -f1,2)
+
+%.pdml1.current:
+	@./scripts/sample_test.sh "$(TSHARK_EXECUTABLE)" "$(basename $(basename $@))" pdml1 $(SELECTED_VERSIONS)
+
+%.pdml2.current:
+	@./scripts/sample_test.sh "$(TSHARK_EXECUTABLE)" "$(basename $(basename $@))" pdml2 $(SELECTED_VERSIONS)
+
+%.text.current:
+	@./scripts/sample_test.sh "$(TSHARK_EXECUTABLE)" "$(basename $(basename $@))" text $(SELECTED_VERSIONS)
+
+all:
```

Comment:
I think that the default target should still be "test" and that this information belongs in the README (or `make help` if you still want a summary).